Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3458,50 +3459,185 @@ def panorama_delete_url_filter_command(url_filter_name: str):
-    if isinstance(rule.get('application'), dict) and 'member' in rule['application']:
-        pretty_rule['Application'] = rule['application']['member']
-    if isinstance(rule.get('destination'), dict) and 'member' in rule['destination']:
-        pretty_rule['Destination'] = rule['destination']['member']
-    if isinstance(rule.get('from'), dict) and 'member' in rule['from']:
-        pretty_rule['From'] = rule['from']['member']
-    if isinstance(rule.get('service'), dict) and 'member' in rule['service']:
-        pretty_rule['Service'] = rule['service']['member']
-    if isinstance(rule.get('to'), dict) and 'member' in rule['to']:
-        pretty_rule['To'] = rule['to']['member']
-    if isinstance(rule.get('source'), dict) and 'member' in rule['source']:
-        pretty_rule['Source'] = rule['source']['member']
-    if isinstance(rule.get('tag'), dict) and 'member' in rule['tag']:
-        pretty_rule['Tags'] = rule['tag']['member']
-    if isinstance(rule.get('log-setting'), dict) and '#text' in rule['log-setting']:
-        pretty_rule['LogForwardingProfile'] = rule['log-setting']['#text']
-    if disabled := rule.get('disabled'):
-        pretty_rule['Disabled'] = disabled
 
-    return pretty_rule
+    rule_get = partial(xml_get, rule)
+
+    tick_box: Dict[Any, str] = {
+        'yes': '✅',
+        'no': '⬜',
+        None: '⬜',
+    }
+
+    context_rule: Dict[str, Any] = {
+        'NegateDestination': rule_get('negate-destination'),
+        'Disabled': rule_get('disabled'),
+        'ICMPUnreachable': rule_get('icmp-unreachable'),
+        'Description': rule_get('description'),
+        'ProfileSetting': rule_get('profile-setting'),
+        'GroupTag': rule_get('group-tag'),
+        'Schedule': rule_get('schedule'),
+        'LogStart': rule_get('log-start'),
+        'LogForwardingProfile': rule_get('log-setting'),
+        'NegateSource': rule_get('negate-source'),
+    }
+    pretty_rule: Dict[str, Any] = {}
+
+    context_rule['DeviceGroup'] = DEVICE_GROUP or None
+
+    context_rule['Location'] = \
+        pretty_rule['Location'] = \
+        rule.get('@loc')
+
+    context_rule['SecurityProfileGroup'] = \
+        pretty_rule['Profile Group'] = \
+        rule_get('profile-setting', subkeys=('group', 'member', '#text'))
+
+    profiles = {
+        k: xml_get(pfs, k)
+        for k in pfs.keys()
+        if not str(k).startswith('@')
+    } \
+        if type(pfs := dict_safe_get(rule, ('profile-setting', 'profiles'))) is dict \
+        else None
+
+    pretty_rule['Profile'] = \
+        context_rule['SecurityProfile'] = \
+        profiles
+
+    target = dict_safe_get(rule, ('target',), {}, dict)
+    target |= {
+        'devices': xml_get(target, 'devices'),
+        'negate': xml_get(target, 'negate'),
+    }
+    context_rule['Target'] = target
+    pretty_rule['Target'] = (
+        f'negate: {target["negate"]}\n'
+        f'devices: {target["devices"]}'
```

Comment:
use `get` of dict instead of this, can fail on `KeyError` if it doesn't exist, lets prevent that